### ISSUE TITLE
[DataLake-Account] Model validation fails with errors

### ISSUE THREAD

**Issue by veronicagg** [Azure/azure-rest-api-specs#772]
> Validating "examples" and "x-ms-examples" in /Users/Veronica/github/openapi-validation-tools/arm-datalake-account/swagger/account.json:
	> Operation: StorageAccounts_Get
		> x-ms-examples
			> Scenario: CreateGetUpdateDeleteTestGET149
				> InnerErrors:
				  1. code: OBJECT_MISSING_REQUIRED_PROPERTY
				     message: Missing required property: accessKey
				     path: properties
				     description: Azure Storage account properties information.

Model from swagger:
```json
    "StorageAccountProperties": {
      "required": [
        "accessKey"
      ],
      "properties": {
        "accessKey": {
          "type": "string",
          "description": "the access key associated with this Azure Storage account that will be used to connect to it."
        },
        "suffix": {
          "type": "string",
          "description": "the optional suffix for the storage account."
        }
      },
      "description": "Azure Storage account properties information."
    },
```

	> Operation: Account_ListByResourceGroup
		> x-ms-examples
			> Scenario: CreateGetUpdateDeleteTestGET840
				> InnerErrors:
				  1. code: OBJECT_MISSING_REQUIRED_PROPERTY
				     message: Missing required property: dataLakeStoreAccounts
				     path: value,1,properties
				     description: The account specific properties that are associated with an underlying Data Lake Analytics account.
				  2. code: OBJECT_MISSING_REQUIRED_PROPERTY
				     message: Missing required property: defaultDataLakeStoreAccount
				     path: value,1,properties
				     description: The account specific properties that are associated with an underlying Data Lake Analytics account.

Model from swagger:
```json
    "DataLakeAnalyticsAccountProperties": {
      "required": [
        "defaultDataLakeStoreAccount",
        "dataLakeStoreAccounts"
      ],
```

Example from .net test recording:
```json
{
  "parameters": {
    "subscriptionId": "53d9063d-87ae-4ea8-be90-3686c3b8669f",
    "resourceGroupName": "rgaba19536",
    "api-version": "2016-11-01",
    "monitor": "true",
    "expanded": "true"
  },
  "responses": {
    "200": {
      "body": {
        "value": [
          {
            "properties": {
              "provisioningState": "Succeeded",
              "state": "Active",
              "endpoint": "testaba18483.azuredatalakeanalytics.net",
              "accountId": "7db18143-e33d-45a0-87b2-356396234446",
              "creationTime": "2016-11-08T04:05:13.3073786Z",
              "lastModifiedTime": "2016-11-08T04:05:42.8264218Z"
            },
            "location": "East US 2",
            "tags": {
              "updatedKey": "updatedValue"
            },
            "id": "/subscriptions/53d9063d-87ae-4ea8-be90-3686c3b8669f/resourceGroups/rgaba19536/providers/Microsoft.DataLakeAnalytics/accounts/testaba18483",
            "name": "testaba18483",
            "type": "Microsoft.DataLakeAnalytics/accounts"
          },
          {
            "properties": {
              "provisioningState": "Succeeded",
              "state": "Active",
              "endpoint": "testaba18483secondacct.azuredatalakeanalytics.net",
              "accountId": "575e9731-9607-42c0-8c87-11c68cc19ae6",
              "creationTime": "2016-11-08T04:05:47.1668081Z",
              "lastModifiedTime": "2016-11-08T04:05:47.1668081Z"
            },
            "location": "East US 2",
            "tags": {
              "updatedKey": "updatedValue"
            },
            "id": "/subscriptions/53d9063d-87ae-4ea8-be90-3686c3b8669f/resourceGroups/rgaba19536/providers/Microsoft.DataLakeAnalytics/accounts/testaba18483secondacct",
            "name": "testaba18483secondacct",
            "type": "Microsoft.DataLakeAnalytics/accounts"
          }
        ]
      }
    }
  }
}
```

	> Operation: Account_List
		> x-ms-examples
			> Scenario: CreateGetUpdateDeleteTestGET939
				> InnerErrors:
				  1. code: INVALID_TYPE
				     message: Expected type object but found type null
				     path: value,4,tags
				     description: Resource tags
Model from swagger:
```json
    "Resource": {
      "description": "The Resource model definition.",
      "properties": {
        "id": {
          "readOnly": true,
          "type": "string",
          "description": "Resource Id"
        },
        "name": {
          "readOnly": true,
          "type": "string",
          "description": "Resource name"
        },
        "type": {
          "readOnly": true,
          "type": "string",
          "description": "Resource type"
        },
        "location": {
          "type": "string",
          "description": "Resource location"
        },
        "tags": {
          "type": "object",
          "additionalProperties": {
            "type": "string"
          },
          "description": "Resource tags"
        }
      },
```

	> Operation: Account_Create
		> x-ms-examples
			> Scenario: CreateGetUpdateDeleteTestPUT1133
				> InnerErrors:
				  1. code: INVALID_TYPE
				     message: Expected type string but found type null
				     path: properties,lastModifiedTime
				     description: the account last modified time.
				  2. code: INVALID_TYPE
				     message: Expected type string but found type null
				     path: properties,creationTime
				     description: the account creation time.
				  3. code: INVALID_TYPE
				     message: Expected type string but found type null
				     path: properties,endpoint
				     description: the full CName endpoint for this account.
				  4. code: INVALID_TYPE
				     message: Expected type string but found type null
				     path: properties,state
				     description: the state of the Data Lake Analytics account.
				  5. code: ENUM_MISMATCH
				     message: No enum match for: null
				     path: properties,state
				     description: the state of the Data Lake Analytics account.

Model from swagger:
```json
    "DataLakeAnalyticsAccountProperties": {
      "required": [
        "defaultDataLakeStoreAccount",
        "dataLakeStoreAccounts"
      ],
      "properties": {
        "provisioningState": {
          "readOnly": true,
          "type": "string",
          "description": "the provisioning status of the Data Lake Analytics account.",
          "enum": [
            "Failed",
            "Creating",
            "Running",
            "Succeeded",
            "Patching",
            "Suspending",
            "Resuming",
            "Deleting",
            "Deleted"
          ],
          "x-ms-enum": {
            "name": "DataLakeAnalyticsAccountStatus",
            "modelAsString": false
          }
        },
        "state": {
          "readOnly": true,
          "type": "string",
          "description": "the state of the Data Lake Analytics account.",
          "enum": [
            "Active",
            "Suspended"
          ],
          "x-ms-enum": {
            "name": "DataLakeAnalyticsAccountState",
            "modelAsString": false
          }
        },
        "defaultDataLakeStoreAccount": {
          "type": "string",
          "description": "the default data lake storage account associated with this Data Lake Analytics account."
        },
        "maxDegreeOfParallelism": {
          "type": "integer",
          "format": "int32",
          "minimum": 1,
          "default": 30,
          "description": "the maximum supported degree of parallelism for this account."
        },
        "queryStoreRetention": {
          "type": "integer",
          "format": "int32",
          "minimum": 1,
          "maximum": 180,
          "default": 30,
          "description": "the number of days that job metadata is retained."
        },
        "maxJobCount": {
          "type": "integer",
          "format": "int32",
          "minimum": 1,
          "default": 3,
          "description": "the maximum supported jobs running under the account at the same time."
        },
        "systemMaxDegreeOfParallelism": {
          "type": "integer",
          "format": "int32",
          "readOnly": true,
          "description": "the system defined maximum supported degree of parallelism for this account, which restricts the maximum value of parallelism the user can set for the account.."
        },
        "systemMaxJobCount": {
          "type": "integer",
          "format": "int32",
          "readOnly": true,
          "description": "the system defined maximum supported jobs running under the account at the same time, which restricts the maximum number of running jobs the user can set for the account."
        },
        "dataLakeStoreAccounts": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/DataLakeStoreAccountInfo"
          },
          "description": "the list of Data Lake storage accounts associated with this account."
        },
        "storageAccounts": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/StorageAccountInfo"
          },
          "description": "the list of Azure Blob storage accounts associated with this account."
        },
        "creationTime": {
          "type": "string",
          "readOnly": true,
          "format": "date-time",
          "description": "the account creation time."
        },
        "lastModifiedTime": {
          "type": "string",
          "readOnly": true,
          "format": "date-time",
          "description": "the account last modified time."
        },
        "endpoint": {
          "type": "string",
          "readOnly": true,
          "description": "the full CName endpoint for this account."
        }
      },
      "description": "The account specific properties that are associated with an underlying Data Lake Analytics account."
    },
```
	> Operation: Account_Create
			> Scenario: SubmitGetListCancelTestPUT1113
				> InnerErrors:
				  1. code: INVALID_FORMAT
				     message: Object didn't pass validation for format date-time: 0001-01-01T00:00:00
				     path: properties,lastModifiedTime
				     description: the account last modified time.
				  2. code: INVALID_FORMAT
				     message: Object didn't pass validation for format date-time: 0001-01-01T00:00:00
				     path: properties,creationTime
				     description: the account creation time.

**Comment by lmazuel** [comment]
@begoldsm ?

**Comment by begoldsm** [comment]
This issue is one where we would rather not maintain two separate objects for requests and responses, where the request requires sensitive information (password, accessKey, etc.) and the response returns the exact same object with the sensitive information missing. Thoughts on how to resolve this issue nicely @amarzavery or @veronicagg ?

**Comment by begoldsm** [comment]
ping on this @amarzavery. Please let me know once there is a good recommendation for solving this issue without requiring the definition of multiple objects. Cascading inheritance might be a good candidate in this case, but let me know if you have other ideas.

**Comment by kirthik** [comment]
@veronicagg  - Could you please help @begoldsm on how to resolve this issue?

**Comment by veronicagg** [comment]
@kirthik We've been ignoring those errors for now from the tool, as this is an issue that needs discussion/making a decision on our side, see https://github.com/Azure/azure-rest-api-specs/issues/790

**Comment by begoldsm** [comment]
@veronicagg this should be resolved by #1429 can you confirm?

**Comment by veronicagg** [comment]
Model validation result from the PR merged above are green, closing.
